src/lib.rs

```diff
@@ -203,17 +197,24 @@ pub use keyutils as default;
 pub mod secret_service;
 #[cfg(all(
     any(target_os = "linux", target_os = "freebsd", target_os = "openbsd"),
-    any(feature = "sync-secret-service", feature = "async-secret-service")
+    any(feature = "sync-secret-service", feature = "async-secret-service"),
+    not(feature = "linux-native")
 ))]
 pub use secret_service as default;
 
 #[cfg(all(
     target_os = "linux",
-    not(any(
-        feature = "linux-native",
-        feature = "sync-secret-service",
-        feature = "async-secret-service"
-    ))
+    any(
```

Comment:
This condition illustrates well what I tried to explain here: https://github.com/hwchen/keyring-rs/issues/214#issuecomment-2365500645.

It just looks wrong and messy. Also not sustainable: how will it look like in few years, with other new keystores?

Reply:
Thanks for the review. I agree this is messy.  Unfortunately, in my experience, dealing with edge cases is always messy, and (as I explain further in #214) having more than one keystore enabled on a given platform is a fairly rare edge case. Using features to specify a single (default) keystore per platform allows the vast majority of clients to write platform-independent code.

As to the question about what this will look like if we integrate a whole bunch of new keystores on one platform, I'm not particularly concerned that's going to happen.  The only reason the `keyutils` platform was integrated in the first place was to allow using this crate on headless Linux boxes where the SecretStore is generally not available.

Hope this helps explain my reasoning.

Reply:
> The only reason the `keyutils` platform was integrated in the first place was to allow using this crate on headless Linux boxes where the SecretStore is generally not available.

Could you explain a bit more, or give an example? I am not sure to understand.

Reply:
The secret service relies on the system to prompt users out of band with their current process, whenever it encounters a locked item or collection. Headless linux installs don't support this, so secret-service can only be used by unlocking the default collection when starting the service (see the docs in the secret-service module for more about this).

Because @landhb and I wanted to have a default keystore available on headless Linux boxes, he contributed and I integrated the keyutils implementation. There was never any expectation that the both would be used at the same time, because if the secret service can be used there's no reason to use the keyutils.

Reply:
Understood, thanks for the details. From the keyutils inline doc:

> Consider the keyring a secure cache and plan for your application to handle cases where the entry is no-longer available in-memory.

For this particular (and unique) case, having both the keyutils and secret service makes sense (hence #222), right?

Reply:
Absolutely! See my review comments on #222, which I think would be a terrific addition to this crate (done as a separate "combo" keystore rather than as a modification of the existing keystore).